lib/pages/chat/chat.dart

```diff
@@ -174,6 +175,9 @@ class ChatController extends State<Chat>
 
   final ValueNotifier<DateTime?> stickyTimestampNotifier = ValueNotifier(null);
 
+  final ValueNotifier<Map<String, CancelToken>> mediaCancelTokenMapNotifier =
+      ValueNotifier({});
```

Comment:
i worried that if user back to chat list, this map becomes to empty again

Reply:
If the user go back to chat list while some uploadings are not done, should those uploads continue or not @hoangdat ?

Reply:
The uploading is still continue, because it's registered under getIt as a singleton.

Reply:
In order to to fix this case, i think you need to have something like a download manager, register it under the getIt, do all the stuff releated to download here